COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md

```diff
@@ -548,6 +548,254 @@ The result is execution of COBOL program CBLDB21 to read the Db2 table and write
+
+**(3):** The DD statement here define where the Enterprise COBOL compiler resides. You will need to check with your system programmer regarding where the compiler is installed on your system. Alongside them, the Language Environment SCEERUN and SCEERUN2 data sets must be included in the concatenation unless they are available in the LNKLST.
+
+**(4):** The SYSUT DD statements define the utility data sets that the compiler will use to process the source program. All SYSUT files must be on direct-access storage devices.
+
+**(5):** The SYSPRINT DD statement defines the data set that receives output from compiler options.
+
+**(6):** The SYSLIN DD statement defines the data set that receives output from the OBJECT compiler option.
+
+**(7):** The SYSIN DD statement defines the data set to be used as input to the job step, or in other words, the source code.
+
+For more information on the input and output data set that the Enterprise COBOL compiler can use, please refer to the [IBM Documentation](https://www.ibm.com/docs/en/cobol-zos/6.3?topic=zos-defining-compiler-input-output).
+
+## Specifying compiler options
+
+The compiler is installed with default compiler options. However, when installing the compiler, the system programmer can fix certain compiler settings. You cannot override any compiler options that are fixed.
+
+For the options that are not fixed, there are several ways in which you can override the default settings:
+- Code them on the PROCESS or CBL statement in the COBOL source code
+- Include them in the JCL when you start the compiler
+- Include them in an SYSOPTF data set, which is one of the input data set the compiler can use
+
+The compiler will then recognize the options in the following order of precedence from highest to lowest:
+1. Fixed installation defaults
+2. Values of the BUFSIZE, OUTDD, SQL and SQLIMS compiler options for the first program in a batch
+3. Options specified on PROCESS (or CBL) statements, preceding the IDENTIFICATION DIVISION
+4. Options specified on the compiler invocation
+5. Installation defaults that are not fixed
+
+The precedence options in an SYSOPTF data set will depend on where the OPTFILE compiler option is specified. For example, if OPTFILE is specified in a PROCESS statement, the SYSOPTF options will supersede the options specified in the compiler invocation.
+
+Note that this order of precedence also determines which options are in effect when there are conflicting or mutually exclusive options.
+
+For a full list of compiler options, please refer to the [IBM Documentation](https://www.ibm.com/docs/en/cobol-zos/6.3?topic=program-compiler-options).
+
+### Specifying options in the PROCESS statement
+
+Within a COBOL program, you can mode most compiler options in the PROCESS statements. We will need to code the statements before the IDENTIFICATION DIVISION header and before any comment lines or compiler-directing statements.
```

Comment:
Did you mean that you can modify most compiler options?

Reply:
Yup, will edit it with the rests of the changes.